Assets/Code/Behaviours/GenerateCity.cs

```diff
@@ -94,6 +117,40 @@ void Start()
         // https://docs.unity3d.com/ScriptReference/Transform-parent.html?_ga=2.51063529.1903334404.1586621367-266296751.1584777261 
     }
 
+    public MapCell GetCellFromPosition(Vector3 position)
+    {
+        var relativePosition = position - bottomLeftCornerPosition;
+        var row = Mathf.FloorToInt(relativePosition.x) / Mathf.RoundToInt(cellSize);
+        var col = Mathf.FloorToInt(relativePosition.z) / Mathf.RoundToInt(cellSize);
+        try
+        {
+            return _map.GetMapCell(row, col);
+        }
+        catch (Exception)
+        {
+            return null;
+        }
+    }
+
+    public bool IsPositionOutOfBounds(Vector3 position)
+    {
+        return (position.x < bottomLeftCornerPosition.x || position.x > bottomLeftCornerPosition.x + Rows * cellSize ||
+                position.z < bottomLeftCornerPosition.z || position.z > bottomLeftCornerPosition.x + Cols * cellSize);
+    }
+
+    void Update()
+    {
+        foreach (var pedestrian in _pedestrians)
+        {
+            var positionOfLookingAhead = pedestrian.transform.position + pedestrian.Direction * _lookAheadDistanceMagnitude;

```

Comment:
This is cool.
I do have an idea for next version.
We could use the Cell's ConnectionPoints (bad name, i know 😅) which represents the relative path (from 0,0 to 1,1) to the center of the cell that this cell containes.

So for example, on a cell that has only 1 direction, you will have two connections points.
one from the center to the first edge, and the other from the center to the other edge.
This helps you know how to move in this cell.
So you could know the direction you need to walk if the cell is a line or it is a left/right turn, or even if it is an intersection which means you will have 3 points or 4 points (depends on how many roads are connected together.
So you will know when and how to turn.

(Not needed now :) )